diff --git a/configure.ac b/configure.ac
index ae78b18f1..1ccfefaed 100644
--- a/configure.ac
+++ b/configure.ac
@@ -275,7 +275,7 @@ case "$host" in
     *)
         save_LDFLAGS="$LDFLAGS"
         LDFLAGS="$LDFLAGS -Wl,--dynamicbase"
-        AC_LINK_IFELSE([AC_LANG_PROGRAM([],[])], [
+        AC_LINK_IFELSE([false], [
         have_dynamicbase=yes
         EXTRA_LDFLAGS="$EXTRA_LDFLAGS -Wl,--dynamicbase"
         ],[])
