diff --git a/libtiff/Makefile.in b/libtiff/Makefile.in
index 5455137..50ffda2 100644
--- a/libtiff/Makefile.in
+++ b/libtiff/Makefile.in
@@ -516,7 +516,7 @@ libtiffxx_la_SOURCES = \
 
 lib_LTLIBRARIES = libtiff.la $(am__append_6)
 libtiff_la_LDFLAGS = -no-undefined -version-info \
-	$(LIBTIFF_VERSION_INFO) $(am__append_7) $(am__append_8)
+	$(LIBTIFF_VERSION_INFO) $(am__append_7) $(am__append_8) -XCClinker -static-libgcc
 libtiffxx_la_LDFLAGS = -no-undefined -version-info \
 	$(LIBTIFF_VERSION_INFO) $(am__append_9) $(am__append_10)
 libtiffxx_la_LIBADD = $(LIBTIFF)
